bot/exts/moderation/slowmode.py

```diff
@@ -66,37 +89,98 @@ async def set_slowmode(
+                await self.original_slowmode_cache.set(channel.id, channel.slowmode_delay)
+            await self.slowmode_expiration_cache.set(channel.id, expiration_time.timestamp())
+
+            self.scheduler.schedule_at(expiration_time, channel.id, self._revert_slowmode(channel.id))
             log.info(
-                f"{ctx.author} tried to set the slowmode delay of #{channel} to {humanized_delay}, "
-                "which is not between 0 and 6 hours."
+                f"{ctx.author} set the slowmode delay for #{channel} to"
+                f"{humanized_delay} which expires in {humanized_duration}."
             )
+            await channel.edit(slowmode_delay=slowmode_delay)
+            await ctx.send(
+            f"{Emojis.check_mark} The slowmode delay for {channel.mention}"
+            f" is now {humanized_delay} and expires in {expiration_timestamp}."
+        )
+        else:
+            if await self.slowmode_expiration_cache.contains(channel.id):
+                await self.slowmode_expiration_cache.delete(channel.id)
+                await self.original_slowmode_cache.delete(channel.id)
+                self.scheduler.cancel(channel.id)
 
+            log.info(f"{ctx.author} set the slowmode delay for #{channel} to {humanized_delay}.")
+            await channel.edit(slowmode_delay=slowmode_delay)
             await ctx.send(
-                f"{Emojis.cross_mark} The slowmode delay must be between 0 and 6 hours."
+                f"{Emojis.check_mark} The slowmode delay for {channel.mention} is now {humanized_delay}."
             )
+        if channel.id in COMMONLY_SLOWMODED_CHANNELS:
+            log.info(f"Recording slowmode change in stats for {channel.name}.")
+            self.bot.stats.gauge(f"slowmode.{COMMONLY_SLOWMODED_CHANNELS[channel.id]}", slowmode_delay)
+
+    async def _reschedule(self) -> None:
+        log.trace("Rescheduling the expiration of temporary slowmodes from cache.")
+        for channel_id, expiration in await self.slowmode_expiration_cache.items():
+            expiration_datetime = datetime.fromtimestamp(expiration, tz=UTC)
+            channel = self.bot.get_channel(channel_id)
+            log.info(f"Rescheduling slowmode expiration for #{channel} ({channel_id}).")
+            self.scheduler.schedule_at(expiration_datetime, channel_id, self._revert_slowmode(channel_id))
+
+    async def _revert_slowmode(self, channel_id: int) -> None:
+        original_slowmode = await self.original_slowmode_cache.get(channel_id)
+        slowmode_delay = time.humanize_delta(seconds=original_slowmode)
+        channel = self.bot.get_channel(channel_id)
+        log.info(f"Slowmode in #{channel} ({channel.id}) has expired and has reverted to {slowmode_delay}.")
+        await channel.edit(slowmode_delay=original_slowmode)
+        await channel.send(
+            f"{Emojis.check_mark} A previously applied slowmode has expired and has been reverted to {slowmode_delay}."
+        )
```

Comment:
Slowmode changes are usually silent. Unless the mods agreed that this is what they want I would send this message to #mods

Reply:
And if i send it to mod-log, should `get_or_fetch_channel` be used here also? Should the mod-log channel be defined as an instance attribute inside __init__ or is it fine to define it inside of the command provided that `get_or_fetch_channel` is used?